### PR TITLE
[Merged by Bors] - feat(InfiniteSum): infinite product of absolute value

### PR DESCRIPTION
**Pull request by b-mehta**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by loefflerd** [comment]
Might it make more sense to define a more general `HasProd.norm` for arbitrary `NormedDivisionRing`s (which would work e.g. for the reals or p-adics too), and then derive the result for C from that `Complex.norm_eq_abs`?

**Comment by b-mehta** [comment]
I have now done this. It requires `NormedField` rather than `NormedDivisionRing`, as that's the generality `norm_prod` is proved in. It also means we have quite a painful new import.
I also added a version for `abs`, also matching the generality `abs_prod` is proved in.

**Comment by loefflerd** [comment]
Ouch, yes, that's a really big import difference; probably too large to live with. Perhaps, rather than adding these lemmas there, it might be better to have a separate, new file just to hold these lemmas – maybe `Mathlib.Topology.Algebra.InfiniteSum.NormedField` or something. 

I'm sorry to put you through so many iterations here.

File: Mathlib/Topology/Algebra/InfiniteSum/Group.lean [hunk 1]
```diff
@@ -3,6 +3,7 @@ Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl
 -/
+import Mathlib.Analysis.Normed.Field.Basic
```
Comment (loefflerd) [inline Mathlib/Topology/Algebra/InfiniteSum/Group.lean:6]: please revert this line

Comment (b-mehta) [inline Mathlib/Topology/Algebra/InfiniteSum/Group.lean:6]:
```suggestion
```

Comment (b-mehta) [reply Mathlib/Topology/Algebra/InfiniteSum/Group.lean:6]: oops, accidental!

**Comment by loefflerd** [comment]
GitHub doesn't support making suggestions on lines which aren't already part of the PR, so I made a tiny extra PR targetted against this one, #18792, with the fixes needed to get this to build.

**Comment by loefflerd** [comment]
I think this is looking good now

maintainer merge

File: Mathlib/Topology/Algebra/InfiniteSum/Order.lean [hunk 2]
```diff
@@ -284,6 +285,21 @@ theorem Set.Finite.of_summable_const [LinearOrderedAddCommGroup α] [Topological
...
 end LinearOrder
 
+section LinearOrderedCommRing
+
+variable [LinearOrderedCommRing α] [TopologicalSpace α] [OrderTopology α] {f : ι → α} {x : α}
+
+nonrec theorem HasProd.abs (hfx : HasProd f x) : HasProd (|f ·|) |x| := by
+  simpa only [HasProd, ← abs_prod] using hfx.abs
+
+theorem multipliable_abs (hf : Multipliable f) : Multipliable (|f ·|) :=
```
Comment (b-mehta) [inline Mathlib/Topology/Algebra/InfiniteSum/Order.lean:295]:
```suggestion
theorem Multipliable.abs (hf : Multipliable f) : Multipliable (|f ·|) :=
```

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**